### ISSUE TITLE
Docs for `reduce` use the outdated VRL regex syntax

### ISSUE THREAD

**Issue by StephenWakely** [vectordotdev/vector#8310]
The docs [here](https://vector.dev/docs/reference/configuration/transforms/reduce/) use 

```toml
starts_when = 'match(.message, /^[^\s]/)'
```

in the example configuration.

The `/.../` syntax for regexes no longer works. The example should now be 

```toml
starts_when = "match(.message, r'^[^\s]')"
```

**Comment by StephenWakely** [comment]
The example should probably also handle `match` being a fallible function.

**Comment by tshepang** [comment]
fixed by 8d1635eb3bf449231d5acd25eb6c942bbaaff3fa, and it also appears to be not fallible
